### ISSUE TITLE
Add 'deduplicate' option to "ensureXXXIndex" on collections

### ISSUE THREAD

**Issue by wsidl** [tariqdaouda/pyArango#114]
On Hash and Skiplist indexes, a "deduplicate" option is available (at least on 3.3.14+ databases). If it's not given, it defaults to True. If this happens, it modifies the documents if duplicates are found which can break an application.

Currently, the only way to create an index with this setting is to create the index directly with the Index class:
```python
from pyArango.index import Index
coll = db['collection_name']
new_index = Index(coll, creationData={
    'type': 'hash', 
    'fields': ['field-1', 'field-2'], 
    'unique': True, 
    'sparse': False, 
    'deduplicate': False
})
coll.indexes['hash'][new_index.infos['id']] = new_index
```

**Comment by wsidl** [comment]
Should also mention that this should be done for all Hash and Skiplist indexes if you do not want to have 'deduplicate' enabled.

**Comment by wsidl** [comment]
Resolved in PR #112
